### ISSUE TITLE
Image owner is not set at upload

### ISSUE THREAD

**Issue by Tazzios** [JoomGalleryfriends/JoomGallery#121]
**Behavior**
When uploaden an image from the backend or the editor button the owner is not set. So i cannot see who had uploaded the picture.
Only when using the menu item user panel for upload the owner gets set.

**Expected**
Evertime and way an image is uploaded the owner is set.

Do i overlook some reason why it is not set?

info:
Joomgallery 3.6.0
Joomla 3.10.11

**Comment by Elfangor93** [comment]
I have also experienced that. But it does not matter for me.
@MrMusic Do you know why JG 3.x is acting like that?

**Comment by MrMusic** [comment]
Hi,

the image owner is only set when uploading in the frontend, as it was thought that only the admin uploads in the backend anyway. If necessary, you can quickly change the owner via batch editing in the backend.

Regards
MrMusic

**Comment by Tazzios** [comment]
I have multiple users that use **the editor button in the frontend** to upload images. I do not always know who has uploaded the image.

I can see if i can make a pull request to add the user to the plugin but joomgallery is a bit more complex then the plugins and modules that i make. :)

**Comment by Elfangor93** [comment]
Never realized that the JoomPlu upload is using the Backend uploader...

**Comment by Tazzios** [comment]
> Never realized that the JoomPlu upload is using the Backend uploader...

Sorry I found out that I`m talking nonsense.
My users can use front **and backend,** I just asked them and they are using the backend.
Did a test on the frontend and then a username is set.

At which joomgallery php files should i look to add also a username from the backend?

**Comment by Tazzios** [comment]
@MrMusic thanks for the direct link.

This is all what needed to always set the user.
```
 // Owner 
 //if($this->_site) 
 //{ 
   $row->owner   = $this->_user->get('id'); 
 //} 
 //else 
 //{ 
 //  $row->owner   = 0; 
 //} 
```

Is there a change it gets in the official release or do I need to apply it as my own mod?

**Comment by MrMusic** [comment]
>Is there a change it gets in the official release or do I need to apply it as my own mod?

I don't think this will come in JoomGallery 3. For backward compatibility a new configuration option would have to be created, etc.

**Comment by Tazzios** [comment]
Thats also fine, I will edit the file at my own installation. It`s not a big deal if I forget the mod for a few days after an update.
